### ISSUE TITLE
Signals framework for solr syncing

### ISSUE THREAD

**Issue by bwhicks** [Princeton-CDH/ppa-django#82]
We need a Solr signals framework to keep the Solr database in sync with changes made through the Django admin.

**Comment by mnaydan** [comment]
- Edits in all content fields were reflected in public search upon refreshing page. (**Question:** Should the volume number (enumeration/chronology) be displayed on the full catalog view? Or is this just for the admin?) 
- Deleted work successfully removed from public search results (tested individual deletion and bulk).
- When I added an individual item to a collection, clicking on the collection filter in the public search included that newly added item in the results. The badge(s) were displayed in the correct place on the individual results (**Note:** These badges in individual search results are clickable, meaning they change color when clicked, but do not do anything. The badges in the filter area retain a green halo after being clicked. Is this on purpose? Doesn't seem right to me. I'll note this on Nick's ticket too.)  
- Collection filter and badges work for bulk added items.
- Collections on public search update successfully when item removed from collection.
- **PROBLEM:** I renamed Brogan to "Brogan Bros" in the admin, the name updated in the filter section on the public site, but now the badge is greyed out and shows no items in that collection because the badges on the individual items did not change; they still say "Brogan." WEIRDER STILL: I then changed "Brogan Bros" to "Kermit" in the admin, and again it successfully updated in the filter section, but NOW the individual item badges say "Brogan Bros"! BUT if you click on a title and go to the catalog record, it says it's in the "Kermit" collection! If you click on "Kermit" there, it says there are no matching works.
- Deleting a collection successfully removes the badge from the filter and individual items. (and now for some reason after doing that I can click on Kermit and it gives me three search results--the ones on hymnology that I had been adding to collections earlier...)

**Comment by rlskoeser** [comment]
@mnaydan is it possible that the collection changes are just taking a little longer than you expect? Changes should be committed after 3 seconds, but it's possible that it will take a little longer than that before it shows up in the public interface.  I _think_ that might explain everything you've flagged above as a possible problem with the collection name updates.  The discrepancies you're reporting relate to some information coming from the database (like the list of collections and the title detail page) and other information coming from Solr (the search results and collection counts).

For your other question: we've discussed the Hathi enumcron before in meetings and we were told not to display it on the public site. Maybe this is something you all can be thinking about as you work on the edition information; maybe there is better information in the MARC record that we could use instead to help people know which edition or volume of something they're look at.

**Comment by mnaydan** [comment]
@rlskoeser Hm, that makes sense. If that is the case, it definitely took longer than 3 seconds -- closer to 30 sec maybe. But even days later, the search results still display the old collection name, while the title detail page displays the new one--but that's the database/Solr discrepancy you're talking about? Is that fixable or just the way it is? (sorry, I'm not totally clear on that)

I guess I was wondering about the enumcron after the Board expressed interest in the edition question, because you're right, it seems related. I'll talk to Meredith about it as we think through editions.

**Comment by rlskoeser** [comment]
@mnaydan no, days later means it's not working. I tried to follow your steps and look at the site and thought it had updated, but I must have misunderstood. I'll have to take another look at the code and see what we missed.

**Comment by mnaydan** [comment]
@rlskoeser this discrepancy is what I'm talking about if it helps:
<img width="1038" alt="screen shot 2018-06-21 at 2 32 15 pm" src="https://user-images.githubusercontent.com/22945600/41738414-fb971d4a-755f-11e8-84ac-9f2610ee4540.png">
<img width="970" alt="screen shot 2018-06-21 at 2 32 27 pm" src="https://user-images.githubusercontent.com/22945600/41738418-ff70fa26-755f-11e8-8971-312b3a369a43.png">

**Comment by rlskoeser** [comment]
@mnaydan thanks, that is helpful - I see that record on the test site now. Some of the items must have gotten missed on the reindex somehow, and they're hard to find because the collection name has changed.

**Comment by thatbudakguy** [comment]
Questions:
- should the collection labels on individual items be links to the collection, or just a "badge" that shows that the item is in that collection?
- do we want to display the enumcron, esp in context of new edition notes/page info?

**Comment by mnaydan** [comment]
@thatbudakguy Let me talk to Meredith about this on Monday.

**Comment by mnaydan** [comment]
@thatbudakguy The collection labels on individual items should just be a "badge," so there should be no interaction for hover or click. The problem is that the gold coloration makes it look like it is clickable. Maybe if it was the same purple as the rest of the record it would look less confusing? 

We do want to display the enumcron. If at all possible, it makes sense to display it on the initial item return (along with publisher, place, year) *and* the item detail page.

**Comment by bwhicks** [comment]
@mnaydan When you do test this, by all means try the set of tests suggested above again, but particularly note the collection name change's results. I believe it should be resolved now.
